File: Php/codice/L09_Libreria/src/Service/LibroServiceImpl.php
<?php 

namespace App\Service;

use App\Repos\LibroRepo;
use App\Service\LibroService;

class LibroServiceImpl implements LibroService {

    private $repo;

    public function __construct(){
        $this->repo = new LibroRepo();
    }

    public function getLibri(){
        return $this->repo->getLibri();
    }

}

File: Java/workspace-fad/Prj15_LibreriaWeb/src/main/java/services/LibroServiceImpl.java
package services;

import java.util.List;

import model.Libro;
import repos.LibroRepo;

public class LibroServiceImpl implements LibroService {

	private LibroRepo repo;
	
	public LibroServiceImpl() {
		this.repo = new LibroRepo();
	}
	
	@Override
	public List<Libro> getLibri() {
		return repo.getLibri();
	}

}
